src/dune_engine/build_system.ml

```diff
@@ -2389,6 +2389,39 @@ let is_target file =
   let+ targets = targets_of ~dir:(Path.parent_exn file) in
   Path.Set.mem targets file
 
+(* Evaluate a rule without building the action dependencies *)
+module Eval_action_builder = Action_builder.Make_exec (struct
+  type fact = unit
+
+  let merge_facts = Dep.Set.union
+
+  let read_file p ~f =
+    let+ _digest = build_file p in
+    f p
+
+  let register_action_deps deps = Memo.Build.return deps
+
+  let register_action_dep_pred g =
+    let+ ps = Pred.eval g in
+    (ps, ())
+
+  let file_exists = file_exists
+
+  let alias_exists = Load_rules.alias_exists
+
+  let execute_action ~observing_facts:_ _act =
+    (* We don't need to execute this action to compute the final action. *)
+    (* jeremiedimino: but maybe we should capture it somehow, it seems relevant
+       to display in the output of [dune rules] *)
+    Memo.Build.return ()
+
+  let execute_action_stdout ~observing_facts:deps act =
```

Comment:
It seems strange that `execute_action` does nothing yet `execute_action_stdout` goes ahead the fires the action. It seems worth adding an explanation here why these two cases are handled so differently.

Reply:
Yes, this is definitely not great. I need to think more about this.